src/logic/contracts/safeContracts.ts

```diff
@@ -74,24 +74,25 @@ export const getSafeMasterContract = async () => {
   return safeMaster
 }
 
-export const getSafeDeploymentTransaction = (safeAccounts, numConfirmations) => {
+export const getSafeDeploymentTransaction = (safeAccounts: string[], numConfirmations: number, safeCreationSalt: number) => {
   const gnosisSafeData = safeMaster.methods
     .setup(safeAccounts, numConfirmations, ZERO_ADDRESS, '0x', DEFAULT_FALLBACK_HANDLER_ADDRESS, ZERO_ADDRESS, 0, ZERO_ADDRESS)
     .encodeABI()
 
-  return proxyFactoryMaster.methods.createProxy(safeMaster.options.address, gnosisSafeData)
+  return proxyFactoryMaster.methods.createProxyWithNonce(safeMaster.options.address, gnosisSafeData, safeCreationSalt)
```

Comment:
DO you know if all the proxyFactory versions have this method implemented?

Reply:
Mmm I'm not sure, @rmeissner could you confirm please?

Reply:
We are gathering that info from the @gnosis.pm/safe-contracts package. I think it's safe to assume that that method will be available.

Reply:
our proxy factory always exposes this method ;)